test/com/google/javascript/jscomp/CollapsePropertiesTest.java

```diff
@@ -233,11 +320,36 @@ public void testAliasCreatedForObjectDepth2_2() {
          "var a$b = {c: 0}; for (var p in a$b) { e(a$b[p]); }");
   }
 
+  public void testEnumDepth1() {
+    test("/** @enum */ var a = {b: 0, c: 1};",
+         "var a$b = 0; var a$c = 1;");
+
+    test("/** @enum */ var a = { /** @nocollapse */ b: 0, c: 1};",
```

Comment:
Enums have two kinds of properties (it may or may not be possible to distinguish between the two kinds in this pass, I'm not sure). The first kind is the properties declared in the enum's object literal. These are the enumerated values, and we should not allow nocollapse there (if possible). The second kind is the properties added after the enum declaration. This is when the enum is treated as a namespace, and we can allow nocollapse for these properties.

Actually, John mentioned to me recently that maybe we should stop supporting enums as namespaces. So, if you can't distinguish between the two kinds of enum properties here, you can even disallow nocollapse on enum properties altogether.